### PR TITLE
VideoPress: connect UI settings with data handling

### LINKED ISSUE #26938: VideoPress: handle isPrivate flag correctly when deciding if a thumbnail is private
The video privacy setting inside the VideoPress library allows 3 values:

1. public
1. private
1. same as site

1 and 2 are intuitive. The third value informs that the video privacy should follow the site settings. This setting is defined inside `Jetpack > Settings`, on the `Performance` tab:

image

The setting is only available when the Jetpack plugin is active.

We need to make sure the changes made to acomodate private thumbnails also include a check on this additional flag.

### Tasks

- [x] Frontend: Implement the UI (https://github.com/Automattic/jetpack/pull/27290)
- [x] Backend: get/update the site option (https://github.com/Automattic/jetpack/pull/27293)
- [x] Data Handling: Connect the backend with front-end by hitting an endpoint
   - [x] use selector&resolver to get the current value for the privacy setting (#27321)
   - [x] connect the current value to the UI component (https://github.com/Automattic/jetpack/pull/27350)
   - [x] update the value on the backend when the value of the component changes using the API (https://github.com/Automattic/jetpack/pull/27350)

### PR DESCRIPTION
**Pull request by retrofox**
Fixes #

#### Changes proposed in this Pull Request:

* VideoPress: connect the UI of the settings section with the data handling

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Go to VideoPress dashboard
* Set the privacy Video Privacy
* Confirm the checkbox toggles
* Confirm the changes are presistent

Screen Shot 2022-11-10 at 14 52 11

###  Network tab

image

image

### State tree
image

### CODE REVIEW

File: projects/packages/videopress/src/client/state/actions.js [hunk 1]
```diff
@@ -357,6 +358,41 @@ const setVideoPressSettings = videoPressSettings => {
...
+const updateVideoPressSettings = settings => async ( { dispatch } ) => {
+	if ( ! settings ) {
+		return;
+	}
+
+	const data = { force: true };
+
+	if ( settings.videoPressVideosPrivateForSite !== undefined ) {
+		data.videopress_videos_private_for_site = settings.videoPressVideosPrivateForSite;
+	}
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/state/actions.js:376]: Maybe a `typeof settings.videoPressVideosPrivateForSite !== 'boolean'` would be a more strict check here.

**Review by dhasilva — APPROVED** [review]
LGTM. Beautiful! :brain:

**Comment by retrofox** [comment]
> LGTM. Beautiful! 🧠

Props to the team, but especially to Luiz.

File: projects/packages/videopress/src/client/state/actions.js (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/state/actions.js:376]: Addressing in a follow-up.

Comment (retrofox) [reply projects/packages/videopress/src/client/state/actions.js:376]: you mean to `if ( settings.videoPressVideosPrivateForSite !== undefined ) {`, do you?

Comment (dhasilva) [reply projects/packages/videopress/src/client/state/actions.js:376]: Yes. Technically, it could be, say, a number.

**Pull request merged**